frb_codegen/src/generator/rust/ty_general_list.rs

```diff
@@ -38,21 +38,31 @@ impl TypeRustGeneratorTrait for TypeGeneralListGenerator<'_> {
         ])
     }
 
-    fn wrap_obj(&self, obj: String, _wired_fallible_func: bool) -> String {
+    fn wrap_obj(&self, obj: String, wired_fallible_func: bool) -> String {
         let inner = TypeRustGenerator::new(
             *self.ir.inner.clone(),
             self.context.ir_file,
             self.context.config,
         );
+
         inner
             .wrapper_struct()
             .map(|wrapper| {
-                format!(
-                    "{}.into_iter().map(|v| {}({})).collect::<Vec<_>>()",
-                    obj,
-                    wrapper,
-                    inner.self_access("v".to_owned())
-                )
+                if wired_fallible_func {
+                    format!(
+                        "{}.map(|s| s.into_iter().map(|v| {}({})).collect::<Vec<_>>())",
+                        obj,
+                        wrapper,
+                        inner.self_access("v".to_owned())
+                    )
+                } else {
+                    format!(
+                        "{}.into_iter().map(|v| {}({})).collect::<Vec<_>>()",
+                        obj,
+                        wrapper,
+                        inner.self_access("v".to_owned())
+                    )
+                }
```

Comment:
uber-nit: maybe we can extract the common code part of the two branches

Reply:
I don't really see a way to do this easily, while remain readability. So I would keep it like this.

Reply:
maybe sth like

```rust
fn f(name: &str) { format!("{}.into_iter().map(|v| {}({})).collect::<Vec<_>>()", name, wrapper, ...) }
if wired_fallible_func {
  format!("{}.map({})", obj, f("s"))
} else {
  f(obj)
}

Reply:
Ok, this should do it.